src/utils/set-bot-commants.ts

```diff
@@ -0,0 +1,81 @@
+import { SetMyCommandsParams } from "../commands.ts";
+import { Api } from "../deps.deno.ts";
+
+const isCommandTooLong = (command: string) => command.length > 32;
+const doesCommandContainInvalidChars = (command: string) =>
+    !/^[a-z0-9_]+$/.test(command);
+
+const isCommandValid = (command: string) => {
+    return !isCommandTooLong(command) &&
+        !doesCommandContainInvalidChars(command);
+};
+
+/**
+ * Options for the `setBotCommands` function.
+ */
+export interface SetBotCommandsOptions {
+    /**
+     * Wether to remove invalid commands from the list of calls to the Bot API.
+     *
+     * If set to `false`, the method will throw an error if any of the commands
+     * is invalid according to the {@link https://core.telegram.org/bots/api#botcommand|official Bot API documentation}.
+     *
+     * Defaults to `false`.
+     */
+    filterInvalidCommands?: boolean;
+}
+
+/**
+ * Performs validation and sets the provided commands for the bot.
+ * @param api Instance of the Api class for the bot the commands are being set for.
+ * @param commandParams List of commands to set.
+ * @param options Options object`
+ */
+export async function setBotCommands(
```

Comment:
Function naming / description should state that this function performs slightly different actions depending on the origin of the `commandParams` input (set commands for the whole Bot or just a single chat)

Reply:
> this function performs slightly different actions depending on the origin of the commandParams

Hm... I don't really agree. It always performs the same actions: validates the commands and calls the Bot API to set them. Wether they'll be set for a specific chat or not is outside of the scope of the function. It makes no difference for it.